apps/ledger-live-desktop/src/internal/transportHandler.ts

```diff
@@ -0,0 +1,108 @@
+import { Subject } from "rxjs";
+import { withDevice } from "@ledgerhq/live-common/hw/deviceAccess";
+
+type APDUMessage = { apduHex: string; requestId: string };
+const transports = new Map<string, Subject<APDUMessage>>();
+
+export const transportOpenChannel = "transport:open";
+export const transportExchangeChannel = "transport:exchange";
+export const transportCloseChannel = "transport:close";
+
+export const transportOpen = ({
+  data,
+  requestId,
+}: {
+  data: { descriptor: string };
+  requestId: string;
+}) => {
+  const subjectExist = transports.get(data.descriptor);
+  if (subjectExist) {
+    // Should we return saying the transport is open ?
+    process.send?.({
+      type: transportOpenChannel,
+      error: new Error("Transport already opened for the given descriptor"),
+      requestId,
+    });
+    return;
+  }
+
+  withDevice(data.descriptor)(transport => {
```

Comment:
you may not need to use withDevice. it's possible what yo just want is to `await` on the generic `open(` because it's only on RENDERER side that we will protect race conditions. you already do a protection with the "one transport at a time"

Reply:
After some digging I think it's better to keep withDevice because without it I would redo a lot of the logic found inside this function